### PR TITLE
Add note about GH Pages redeploy

### PR DESCRIPTION
**Pull request by TheJaredWilcurt**
I changed `master` to `main` on a `username.github.io` repo and the site went to 404. I assumed it was DNS related and would solve itself and went to sleep (using a CNAME file). The site was still down in the morning, so I edited the `index.html` to add a single character change and merged to trigger a deploy. This fixed it.

I'm shocked that GitHub doesn't just perform a one-time deploy of GH Pages after a branch rename. Especially in a scenario where you're going from `master` to `main` on a repo ending in `.github.io`. But really it should just do that anytime the branch used for GH Pages is renamed. Forcing all users to know about this weird edge case and having to add a space to a file to fix it is very hacky.

Until this can be automated by GitHub staff, it should at least be documented.

### CODE REVIEW

**Comment by infin8x** [comment]
@willsmythe can you comment on this?

**Review by queilawithaQ — APPROVED** [review]

**Review by zbianca — APPROVED** [review]

**Comment by TheJaredWilcurt** [comment]
There is now a note added to the alert when you go to rename your branch. Though this is pretty easily missed and it would still be useful to have this documented for those *after* the fact.

screenshot

**Review by devkadirselcuk — APPROVED** [review]

**Review by devkadirselcuk — APPROVED** [review]

**Comment by TheJaredWilcurt** [comment]
Fun fact! I just changed that branch from `master` to `main` 2 days ago, took a screenshot specifically for this issue and then TOTALLY FORGOT to "make a commit adding a single space" to cause a redeploy. So my portfolio site has been down for the last two days.

**Review by nickmccurdy — APPROVED** [review]

**Comment by michaelmior** [comment]
Instead of adding a single space (or some other meaningless) for this, you can just use the `--allow-empty` to `git commit` to allow an empty commit which when pushed will also trigger a redeploy. When I'm the only one working on a project, sometimes I'll also use `git commit --amend` which rewrites the last commit, but only changes the timestamp and then force push this.

**Review by jFJ13 — APPROVED** [review]

**Review by GrpeApple — APPROVED** [review]

**Review by Fizie92 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ We're making changes to GitHub in a few phases, designed to cause as little disr
 - [We've updated GitHub.com](https://github.blog/changelog/2020-07-17-links-to-deleted-branches-now-redirect-to-the-default-branch/) to redirect links that contain the deleted branch name such as `master` to the corresponding link in the repository's new default branch.
 - [We've updated GitHub Pages](https://github.blog/changelog/2020-07-31-build-and-deploy-github-pages-from-any-branch-beta/) to build and deploy from any branch.
   - Note: publishing to the special `gh-pages` branch will still work the same as it always has, but now you can choose any other branch in your repository as the publishing source.
+  - Note: If a branch used for GitHub Pages is renamed, it will not automatically redeploy your site. You will need to edit a file on that branch after the rename to trigger a new build and deploy. Until then your website will direct to a 404 page.
```
Comment (willsmythe) [inline README.md:14]:
```suggestion
  - Note: If the branch used by GitHub Pages is renamed, your GitHub Pages site will be unpublished. Pushing a new commit to the renamed branch will republish the site.
```

**Pull request merged**